### ISSUE TITLE
d3-pack throws error at extendBasis for the given data

### ISSUE THREAD

**Issue by aarthi0808** [d3/d3-hierarchy#150]
Facing same kind of issue in using d3-hierarchy for pack layout. Please check the data i am using,
`             ["a",5.473061352023348e+21],
              ["b", 1.1729457180068344e+21],
               ["c", 1023]`

Checkout the error stack messages

> 
    at extendBasis (d3-hierarchy.v1.js:353)
    at enclose (d3-hierarchy.v1.js:321)
    at packEnclose (d3-hierarchy.v1.js:536)
    at d3-hierarchy.v1.js:626
    at Node.node_eachAfter [as eachAfter] (d3-hierarchy.v1.js:140)
    at pack (d3-hierarchy.v1.js:586)

Need a fix for this issue @mbostock 

_Originally posted by @aarthi0808 in https://github.com/d3/d3-hierarchy/issues/100#issuecomment-533491060_

**Comment by aarthi0808** [comment]
@mbostock Please respond to this issue!

**Comment by mbostock** [comment]
This is an issue of [non-robust arithmetic](https://observablehq.com/@mourner/non-robust-arithmetic-as-art). There’s no simple fix for this.

**Comment by mbostock** [comment]
Here is some code that reproduces the error:

```js
d3.packSiblings([
  {r: 5.473061352023348e+12},
  {r: 1.1729457180068344e+12},
  {r: 1023}
])
```

@robinhouston and I spent some time looking at this before, and perhaps this is my fault for tweaking the enclosure tests. I’d like to fix this, but I’m not entirely sure how to do it. And even if we wanted to adopt [robust-predicates](https://github.com/mourner/robust-predicates), I’m not entirely sure how we’d apply the incircle test here (and also, adding that as a dependency would be significant and likely make this layout slower where a fast approximate solution would suffice).

**Comment by theiliad** [comment]
> This is an issue of [non-robust arithmetic](https://observablehq.com/@mourner/non-robust-arithmetic-as-art). There’s no simple fix for this.

Hey @mbostock, @aarthi0808. I found a fix for this issue in https://github.com/carbon-design-system/carbon-charts

Going through https://observablehq.com/@mourner/non-robust-arithmetic-as-art

I realized that if the chart dimensions `[width, height]` are smaller than 1px, that could fully throw off the math...

Now you might not intend to have a chart at <1px width & height, I didn't either, but for some reason in 1 of the frames in Firefox the chart is actually much smaller than that, `almost 9e-9px`.

So I just wait in my renders until the chart dimensions are larger than 1px
![image](https://user-images.githubusercontent.com/14989804/123330194-b8664a00-d50b-11eb-93b0-8222954b0738.png)

and voila! it works
